src/System.Collections.NonGeneric/src/System/Collections/ArrayList.cs

```diff
@@ -320,17 +315,18 @@ public virtual bool Contains(Object item)
             if (item == null)
             {
                 for (int i = 0; i < _size; i++)
+                {
                     if (_items[i] == null)
                         return true;
+                }
                 return false;
             }
-            else
+            for (int i = 0; i < _size; i++)
```

Comment:
We should avoid removing these else clauses while technically correct right now, it unnecessarily churns the code and sets us up for potential issues later. For example right now all the places in that if block return but someone could end up adding a branch that doesn't return and then this code would be incorrect and cause a subtle bug.